e2etest/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,79 @@
+package metrics
+
+import (
+	"fmt"
+	"net"
+	"path"
+	"strings"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/pkg/errors"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// MetricsForPod returns the parsed metrics for the given pod, scraping them
+// from the executor pod.
+func ForPod(executor, target *corev1.Pod) (map[string]*dto.MetricFamily, error) {
+	metricsUrl := path.Join(net.JoinHostPort(target.Status.PodIP, "7472"), "metrics")
```

Comment:
Is this a standard port? If not, is there any way to determine it?

Reply:
It depends on the deployment (meaning, the deployment done via manifests currently uses that port, but it could change).
We could inspect the pod checking what pod is using, but I am not sure it's worth the effort. Another option is to have this overrideable from outside, but I'd wait to have the problem. If that helps, I can have the port as a constant.